### ISSUE TITLE
Ideas for future functionality

### ISSUE THREAD

**Issue by Kimax89** [nicehash/NiceHashMiner-Archived#817]
I like how easy and straight forward the NiceHashMiner is to use, but i dont think that the miner appeals to miners with more than 1 rig. So i came up with a few ideas that could benefit alot of users.

1. It would be nice if the software itself had API access, that would allow users to monitor and control the software. Monitoring the hashrate through an API would allow for scripting and users to quickly identify potential faulty hardware. And if this API ever comes to life, then please make it as flexible as possible. Just like the extended provider API on the website. I really love that API.

2. Remote Control and Monitoring Software. By using the API i just mentioned. This software would be for the bigger miners that uses more than 1 rig and wants and easy-to-use monitoring software. From here you would be able to see the hashrate of all the miners that are connected and be able to start/stop, change the algorithm, do a new benchmark. Basically do everything you would be able to do in the NiceHashMiner, but here, you are doing it remotely.

That was all from me. I really hope to see this being implemented in the future, as i believe it would benefit ALOT of users and making this software go from appealing to hobby miners, to industry standart.

Keep up the good work and i cant wait to see what happens in the future of Bitcoins and Bitmining.

**Comment by Joshwaa2010** [comment]
Yes please implement something for multi-rig control.  I have 8 rigs and this would make things much easier.

**Comment by DillonN** [comment]
I've also wanted this for a while, it's been on my back burner for things to work on. If there isn't official support in a couple weeks I will probably try making my own implementation and put up a pull request for it

**Comment by aidonaks** [comment]
> Yes please implement something for multi-rig control. I have 8 rigs and this would make things much easier.

I agree too. Have one 7-GPU rig today, already bought 2 more which will be running in 10 days, plus a fourth by mid-June, and finally at least 10 more by July. A total of  at least 14+ rigs by July end - All Windows 10. Would love to have a way to manage them remotely without having to open 14 teamviewer sessions. If not manage, at least monitor them remotely in a dashboard, which can alert me to an issue on say Rig #7. I can then open a TV session for that rig to fix it.

Issues like the PC hanging, or restarting are fixed easily, **but what about issues like the miner hanging/freezing/giving errors while the PC continues to runs smoothly**? Something like this in v2 of NHM would really help.

**Comment by Kimax89** [comment]
@S74nk0 Do you have any comments on this?

**Comment by Valinwolf** [comment]
About a year without response, yet it's still open. Is this a feature or planned feature?

**Comment by Kimax89** [comment]
@Tinkerwolf I really hope so.

**Comment by Valinwolf** [comment]
@Kimax89 If it is already a feature, I couldn't find docs on it. So, I assume it has not been implemented. My problem is I want to develop a safe guard program that disables things or even shuts down based on temperature, so that I can leave my PC unattended without worrying about my CPU/GPU overheating and dying. I didn't want the NHM to permanently stop (killing it) on overheat, just let it have an hour of rest to cool off.
